src/coreclr/jit/gentree.h

```diff
@@ -1292,9 +1292,9 @@ struct GenTree
 
     bool OperIsPutArgSplit() const
     {
-#if FEATURE_ARG_SPLIT
-        return gtOper == GT_PUTARG_SPLIT;
-#else // !FEATURE_ARG_SPLIT
+#if FEATURE_ARG_SPLIT_SUPPORTED
+        return compFeatureArgSplit() && (gtOper == GT_PUTARG_SPLIT);
```

Comment:
This is another case where it seems like we shouldn't have to check `compFeatureArgSplit`; we should just ensure we don't create a GT_PUTARG_SPLIT if it's not legal.